modules/fluid_properties/src/fluidproperties/IdealRealGasMixtureFluidProperties.C

```diff
@@ -14,8 +14,224 @@
+  Real IdealRealGasMixtureFluidProperties::prop##_from_v_e(                                        \
+      Real v, Real e, const std::vector<Real> & x) const                                           \
+  {                                                                                                \
+    const Real T = T_from_v_e(v, e, x);                                                            \
+    return prop##_from_T_v(T, v, x);                                                               \
+  }                                                                                                \
+                                                                                                   \
+  void IdealRealGasMixtureFluidProperties::prop##_from_v_e(Real v,                                 \
+                                                           Real e,                                 \
+                                                           const std::vector<Real> & x,            \
+                                                           Real & y,                               \
+                                                           Real & dy_dv,                           \
+                                                           Real & dy_de,                           \
+                                                           std::vector<Real> & dy_dx) const        \
+  {                                                                                                \
+    Real T, dT_dv, dT_de;                                                                          \
+    std::vector<Real> dT_dx(_n_secondary_vapors);                                                  \
+    T_from_v_e(v, e, x, T, dT_dv, dT_de, dT_dx);                                                   \
+                                                                                                   \
+    Real dy_dT_v, dy_dv_T;                                                                         \
+    std::vector<Real> dy_dx_Tv;                                                                    \
+    prop##_from_T_v(T, v, x, y, dy_dT_v, dy_dv_T, dy_dx_Tv);                                       \
+                                                                                                   \
+    dy_dv = dy_dv_T + dy_dT_v * dT_dv;                                                             \
+    dy_de = dy_dT_v * dT_de;                                                                       \
+    dy_dx.resize(_n_secondary_vapors);                                                             \
+    for (unsigned int i = 0; i < _n_secondary_vapors; i++)                                         \
+      dy_dx[i] = dy_dx_Tv[i] + dy_dT_v * dT_dx[i];                                                 \
+  }
+
+/**
+ * Defines the value and derivative methods from (p,T) for a property y
```

Comment:
this is pretty general. it does not have to be an IdealRealGas right? or is the lack of interaction between phase the definition of "IdealReal:" ?

Reply:
It could potentially be moved up into `VaporMixtureFluidProperties`

Reply:
Actually nevermind, the (T,v) interfaces are not in `VaporMixtureFluidProperties`, so it belongs here only.